core/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -846,14 +887,13 @@ object AutoTuner extends Logging {
   val DEF_WORKER_GPU_NAME = "T4"
   // T4 default memory is 16G
   // A100 set default to 40GB
-  val DEF_WORKER_GPU_MEMORY_MB: mutable.LinkedHashMap[String, String] =
-    mutable.LinkedHashMap[String, String]("T4"-> "15109m", "A100" -> "40960m")
+  val DEF_WORKER_GPU_MEMORY_MB: Map[String, String] = Map("T4"-> "15109m", "A100" -> "40960m")
```

Comment:
I am a bit confused with the comment and actual assignment of memory for T4. For A100, the conversion between MB to GB seems fine. Is this intentional for T4?

Reply:
Yes, the value here is less than 16G for T4. Back then, we looked-up the memory from the dataproc clusters.
For some reason it does not use the full 16G (maybe accounting for overhead)
We likely going to revisit that when we fully support the GPU-devices like L4s. It was not in the scope to make that change in this PR.

P.S: I did not change that in this PR. I fixed removing a mutable Hash.